src/vaadin-combo-box-dropdown-wrapper.html

```diff
@@ -168,7 +199,41 @@
       }
 
       _getItems(opened, items) {
-        return opened ? items : [];
+        if (opened) {
+          if (this._notEmpty(items) && !this.filterChanged) {
+            // iron-list triggers the scroller's reset after items update, and
+            // this is not appropriate for undefined size lazy loading.
+            // see https://github.com/vaadin/vaadin-combo-box-flow/issues/386
+            // We store iron-list scrolling position in order to restore
+            // it later on after the items have been updated.
+            const currentScrollerPosition = this._selector.firstVisibleIndex;
+            if (currentScrollerPosition !== 0) {
+              this._oldScrollerPosition = currentScrollerPosition;
+              this.resetScrolling = true;
+            }
+          }
+          // Let the position to be restored in the future calls unless it's not
+          // caused by filtering
+          this.filterChanged = false;
+          return items;
+        }
+        return [];
+      }
+
+      _restoreScrollerPosition(items) {
+        if (this._notEmpty(items) && this._oldScrollerPosition !== 0) {
+          // new items size might be less than old scrolling position
+          this._selector.scrollToIndex(Math.min(items.length - 1, this._oldScrollerPosition));
+          this.resetScrolling = false;
+          // reset position to 0 again in order to properly handle the filter
+          // cases (scroll to 0 after typing the filter)
+          this._oldScrollerPosition = 0;
+        }
+      }
+
+      _notEmpty(items) {
```

Comment:
Given the signature of this function, you'd expect it to check if the provided array is not empty. But it also seems to rely on a state of a global `this._selector` variable, which is a bit unexpected.

Could change it to only check for the array: `return items && items.length`. If this new helper is added to the module, it could also be used [here](https://github.com/vaadin/vaadin-combo-box/blob/master/src/vaadin-combo-box-dropdown-wrapper.html#L184) and [here](https://github.com/vaadin/vaadin-combo-box/blob/master/src/vaadin-combo-box-dropdown-wrapper.html#L472)

Reply:
Done